src/preprocessing/passes/rewrite.cpp

```diff
@@ -27,15 +28,24 @@ namespace passes {
 using namespace cvc5::internal::theory;
 
 Rewrite::Rewrite(PreprocessingPassContext* preprocContext)
-    : PreprocessingPass(preprocContext, "rewrite"){};
-
+    : PreprocessingPass(preprocContext, "rewrite"),
+      d_proof(options().smt.produceProofs ? new RewriteProofGenerator(d_env)
+                                          : nullptr)
+{
+}
 
 PreprocessingPassResult Rewrite::applyInternal(
   AssertionPipeline* assertionsToPreprocess)
 {
   for (size_t i = 0, size = assertionsToPreprocess->size(); i < size; ++i)
   {
-    assertionsToPreprocess->replace(i, rewrite((*assertionsToPreprocess)[i]));
+    const Node& a = (*assertionsToPreprocess)[i];
+    Node ar = rewrite(a);
+    if (a == ar)
+    {
+      continue;
+    }
```

Comment:
the `replace` method is already testing whether the original assertion and the result of rewriting are the same. Why add this here again? Same for the case in the other pass.

Reply:
Ive simplified.